lecture_02/assignment_01/francesco_milano/assignment_01.py

```diff
@@ -0,0 +1,33 @@
+from compas.artists import Artist
+from compas.datastructures import Mesh
+from compas.geometry import Box
+from compas.geometry import Frame
+from compas.geometry import Plane
+from compas.geometry import Projection
+
+# construct a frame
+frame = Frame([0, 0, 200], [0, 1, 0], [1, 0, 0])
+
+#construct a box
+width, length, height = 200, 100, 100
+box = Box(frame, width, length, height)
+
+# transformatin matrix
+point = [0, 0, 0]
+normal = [0, 0, 1]
+plane = Plane(point, normal)
+direction = [1, 1, 1]  #
+P = Projection.from_plane_and_direction(plane, direction)
+
+# box to mesh
+mesh = Mesh.from_shape(box)
+
+# apply the projection onto the mesh
+mesh_projected = Mesh.transformed(mesh, P)
+# why transformED?
```

Comment:
There are two methods to transform an object: `transform()` and `transformed()`, the past-tense version applies the transformation to a new (copy) of the mesh, while the other one makes the changes in-place modifying the `mesh` argument directly.

You used it as if it were a static method, `Mesh.transformed(mesh, P)` instead of an instance method: `mesh.transformed(P)`. Both usages are valid, but the latter is a bit shorter and nicer to read.